tests/test_app.py

```diff
@@ -8,6 +9,20 @@ def test_init():
     assert app
 
 
+def test_run():
+    app = Application()
+    app.run_internally = mock.MagicMock(return_value=True)
+    assert app.run()
+    assert app.run_internally.called
+
+
+def test_run_with_exception():
```

Comment:
Why is your above way, defining run_internally manually for the test is than using mock?
Using mock looks better for  me.

```
app.run_internally = mock.Mock(side_effect=RuntimeError(...))
```